pkg/backup/client.go

```diff
@@ -423,10 +423,12 @@ func (bc *Client) BackupRanges(
 	// we collect all files in a single goroutine to avoid thread safety issues.
 	workerPool := utils.NewWorkerPool(concurrency, "Ranges")
 	eg, ectx := errgroup.WithContext(ctx)
-	for _, r := range ranges {
+	for id, r := range ranges {
+		id := id
```

Comment:
Is it redundant?

Reply:
Nope, we need make `id` a local variable so the closure won't capture its change, which is a common trap in Go🤯.

See [this playground](https://play.golang.org/p/OyMjs3Bl4su).
```go
const max = 6

func main() {
	wg := new(sync.WaitGroup)
	wg.Add(max)
	for i := 0; i < max; i++ {
		go func() { fmt.Printf("%d", i); wg.Done() }()
	}
	wg.Wait()
	// output: 656666
}
```